tests/rptest/tests/topic_recovery_test.py

```diff
@@ -1064,6 +1064,59 @@ def _wipe_data(self):
                 f"All data will be removed from node {node.account.hostname}")
             self.redpanda.remove_local_data(node)
 
+    def _wait_for_data_in_s3(self,
+                             expected_topics,
+                             timeout=datetime.timedelta(minutes=1)):
+        """Wait until all topics are uploaded to S3"""
+        deadline = datetime.datetime.now() + timeout
+        while True:
+            try:
+                total_partitions = sum(
+                    [t.partition_count for t in expected_topics])
+                manifests = []
+                topic_manifests = []
+                segments = []
+                lst = self.s3_client.list_objects(self.s3_bucket)
+                for obj in lst:
+                    if obj.Key.endswith("/manifest.json"):
+                        manifests.append(obj)
+                    elif obj.Key.endswith("/topic_manifest.json"):
+                        topic_manifests.append(obj)
+                    else:
+                        segments.append(obj)
+                if len(expected_topics) != len(topic_manifests):
+                    self.logger.info(
+                        f"can't find enough topic_manifest.json objects, expected: {len(expected_topicsi)}, actual: {len(topic_manifest)}"
+                    )
+                    raise ValueError()
```

Comment:
Blank ValueErrors are a bit unusual, maybe this can just a "return False" if when this is made into a helper function with wait_until?